File: src/main/java/no/novari/flyt/mapping/kafka/configuration/KafkaRequestReplyProperties.java
package no.novari.flyt.mapping.kafka.configuration;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.time.Duration;

@Getter
@Setter
@Component
@ConfigurationProperties(prefix = "fint.flyt.mapping-service.kafka.request-reply")
public class KafkaRequestReplyProperties {

    private Duration replyTimeout;
    private Duration replyRetentionTime;
}

File: src/main/kotlin/no/novari/flyt/mapping/kafka/configuration/KafkaRequestReplyProperties.kt
package no.novari.flyt.mapping.kafka.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import java.time.Duration

@Component
@ConfigurationProperties(prefix = "novari.flyt.mapping-service.kafka.request-reply")
class KafkaRequestReplyProperties {
    var replyTimeout: Duration = Duration.ofSeconds(5)
    var replyRetentionTime: Duration = Duration.ofMinutes(5)
}
